src/vsr/replica.zig

```diff
@@ -731,10 +731,24 @@ pub fn ReplicaType(
                 assert(trailer_chunks.len == 1);
                 assert(trailer_size == ClientSessions.encode_size);
                 assert(trailer_size == trailer_chunks[0].len);
-
                 self.client_sessions.decode(trailer_chunks[0]);
             }
 
+            maybe(self.client_replies.writing.count() > 0);
+            for (0..constants.clients_max) |entry_slot| {
+                if (self.client_sessions.entries_free.isSet(entry_slot)) {
+                    assert(!self.client_replies.faulty.isSet(entry_slot));
+                } else {
+                    const entry = &self.client_sessions.entries[entry_slot];
+                    if (entry.header.op >= self.superblock.working.vsr_state.sync_op_min and
```

Comment:
Wait, we also run this during _normal_ open, I guess we need to do something different for that case...

Reply:
Gated this with "if state-syncing". As a nice bonus, this sets faulty bits correctly after replica restarts _during_ state sync.